### PR TITLE
Handle exception on move without internet connection

### PR DESCRIPTION
**Pull request by beerline**
Hi **flutter_map** team!

Here is desigion for issue #593 

- add errorTileCallback for handling errors while no tiles
- example how to use errorTileCallback

Main goal to catch this exception, is to notify user about internet connection lost.

### CODE REVIEW

File: example/lib/widgets/drawer.dart [hunk 1]
```diff
@@ -156,6 +157,13 @@ Drawer buildDrawer(BuildContext context, String currentRoute) {
             Navigator.pushReplacementNamed(context, OverlayImagePage.route);
           },
         ),
+        ListTile(
+          title: const Text('Tile loading error handle'),
+          selected: currentRoute == OverlayImagePage.route,
```
Comment (avioli) [inline example/lib/widgets/drawer.dart:162]: Shouldn’t this be `TileLoadingErrorHandle.route`?

File: example/lib/pages/tile_loading_error_handle.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        child: Column(
+          children: [
+            Padding(
+              padding: EdgeInsets.only(top: 8.0, bottom: 8.0),
+              child: Text('Turn on Airplane mode and try to move or zoom map'),
+            ),
+            Flexible(
+              child: Builder(builder: (BuildContext context) {
+                return FlutterMap(
+                  mapController: MapController(),
```
Comment (avioli) [inline example/lib/pages/tile_loading_error_handle.dart:33]: This is a bad example use `MapController` use. A map controller is not necessary.

Comment (avioli) [inline example/lib/pages/tile_loading_error_handle.dart:33]: Sorry for nitpicking - this is a bad example use MapController use. A map controller is not necessary.

File: lib/src/layer/tile_layer.dart [hunk 3]
```diff
@@ -181,7 +186,8 @@ class TileLayerOptions extends LayerOptions {
       // Tiles fade in duration in milliseconds (default 100),
       // it can 0 to avoid fade in
       int tileFadeInDuration = 100,
-      rebuild})
+      rebuild,
+      this.errorTileCallback,})
```
Comment (avioli) [inline lib/src/layer/tile_layer.dart:190]: Another nitpick - usually `rebuild` is last.

**Comment by beerline** [comment]
i fixed all issues

**Comment by avioli** [comment]
@beerline I’m sorry if I gave you the idea I’m somehow able to merge this pull request into this repo. I was just pointing the issues I saw.

Apparently there is a formatting issue in `drawer.dart` that needs fixing.

@johnpryan can merge it, after reviewing if for himself, but it is up to him to find the time to do so and make a choice if the project requires this feature at all.

**Comment by johnpryan** [comment]
This is exposing some weird quirks of the API. Tile is meant to be an internal class, so this callback's API might be subject to change in the future. But I think overall it makes sense to provide a way to catch these problems in the app.

**Comment by johnpryan** [comment]
Thanks for reviewing @avioli

**Pull request merged**